vault.gemspec

```diff
@@ -19,6 +19,8 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
+  spec.add_runtime_dependency "aws-sdk"
```

Comment:
It _should_ just be hmacing a bunch of things.

```ruby
def aws_signature(key, timestamp, region, service)
  k_date    = OpenSSL::HMAC.digest('sha256', "AWS4" + key, timestamp)
  k_region  = OpenSSL::HMAC.digest('sha256', k_date, region)
  k_service = OpenSSL::HMAC.digest('sha256', k_region, service)
  k_signing = OpenSSL::HMAC.digest('sha256', k_service, "aws4_request")
  kSigning
end
```

This looks to be a more complete example: https://gist.github.com/blelump/9e6880a87fefa6397032